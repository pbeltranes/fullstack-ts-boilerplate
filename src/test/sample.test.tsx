import { describe, test } from 'vitest';
import Page from '@/app/page';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';

describe('SampleComponent', () => {
    test('renders correctly', () => {
        render(<Page />);
        // Añade tus aserciones aquí
    });

    test('displays the correct title', () => {
        render(<Page />);
        const titleElement = screen.getByText(/Content Section/i);
        expect(titleElement).toBeInTheDocument();
    });

    test('renders a button', () => {
        render(<Page />);
        const buttonElement = screen.getByRole('button');
        expect(buttonElement).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<Page />);
        expect(asFragment()).toMatchSnapshot();
    });
});