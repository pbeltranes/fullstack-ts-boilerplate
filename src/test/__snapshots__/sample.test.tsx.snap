// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SampleComponent > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-12"
  >
    <h1
      class="text-4xl font-bold mb-6"
    >
      Welcome to My Site
    </h1>
    <p
      class="text-xl mb-8"
    >
      This is a simple one-page layout with a responsive navbar and theme toggle.
    </p>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <div
          class="text-2xl font-semibold leading-none tracking-tight"
        >
          Content Section
        </div>
        <div
          class="text-sm text-muted-foreground"
        >
          Here you can add more content for your single-page website.
        </div>
      </div>
      <div
        class="p-6 pt-0"
      >
        <p>
          This could include features, testimonials, or any other relevant information.
        </p>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mt-4"
        >
          Learn More
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
